app/src/main/java/com/example/farmeraid/farm/FarmViewModel.kt

```diff
@@ -82,12 +84,16 @@ class FarmViewModel @Inject constructor(
 
     init{
         viewModelScope.launch {
-            inventoryRepository.getInventory().collect{ produce ->
-                harvestList.value = produce.map {(produceName, _) ->
-                    FarmModel.ProduceHarvest(
-                        produceName = produceName,
-                        produceCount = 0,
-                    )
+            userRepository.getUserId()?.let {
```

Comment:
Now that youre doing the user validation check in the repositories, you should need to check userRepository in this viewmodel.